### ISSUE TITLE
Not able to compile

### ISSUE THREAD

**Issue by WitherCubes** [codic12/worm#21]
# Sorry if there is any mistake on my side, but I can't compile worm and it shows an error while doing it.

Here is the full output when running `nimble build -d:release --gc:arc`:

```
  Verifying dependencies for worm@0.1.0
      Info: Dependency on x11@any version already satisfied
  Verifying dependencies for x11@1.1
   Building worm/wormc using c backend
   Building worm/worm using c backend
/home/wc/.cache/nim/worm_r/@mworm.nim.c: In function ‘getProperty__9aiI5srh3lf4yinTF7HA9aGQ’:
/home/wc/.cache/nim/worm_r/@mworm.nim.c:3022:26: error: incompatible types when assigning to type ‘tyObject_Option__hcmjwL9c5ECYpHnVevXHwEQ’ from type ‘tyObject_Option__9bcZreTqxpOXg5u4xcKsUEw’
 3022 |                 result = colontmpD__2;
      |                          ^~~~~~~~~~~~
/home/wc/.cache/nim/worm_r/@mworm.nim.c: In function ‘handleUnmapNotify__IBNx9c2pqwde9cI9bE9afvcRpw’:
/home/wc/.cache/nim/worm_r/@mworm.nim.c:4423:27: error: incompatible types when assigning to type ‘tyObject_Option__hcmjwL9c5ECYpHnVevXHwEQ’ from type ‘tyObject_Option__9bcZreTqxpOXg5u4xcKsUEw’
 4423 |         (*self).focused = T12_;
      |                           ^~~~
/home/wc/.cache/nim/worm_r/@mworm.nim.c: In function ‘handleDestroyNotify__UtpRh9a9bdEb3vV3vfdWHAsw’:
/home/wc/.cache/nim/worm_r/@mworm.nim.c:4509:27: error: incompatible types when assigning to type ‘tyObject_Option__hcmjwL9c5ECYpHnVevXHwEQ’ from type ‘tyObject_Option__9bcZreTqxpOXg5u4xcKsUEw’
 4509 |         (*self).focused = T12_;
      |                           ^~~~
Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -O3 -fno-strict-aliasing -fno-ident   -I/usr/lib/nim -I/home/wc/tmp/worm/src -o /home/wc/.cache/nim/worm_r/@mworm.nim.c.o /home/wc/.cache/nim/worm_r/@mworm.nim.c' failed with exit code: 1


       Tip: 5 messages have been suppressed, use --verbose to show them.
     Error: Build failed for package: worm
        ... Execution failed with exit code 1
        ... Command: /usr/bin/nim c --colors:on --noNimblePath -d:release --gc:arc -d:NimblePkgVersion=0.1.0 --path:/home/wc/.nimble/pkgs/x11-1.1 --hints:off -o:/home/wc/tmp/worm/worm /home/wc/tmp/worm/src/worm.nim
```

**Comment by moson-mo** [comment]
It seems you need at least Nim v1.6.0 to be able to build.  

@codic12 Should probably be changed here:
https://github.com/codic12/worm/blob/47ebe569a86358ab3ca17d6ee848316f802a5012/worm.nimble#L13

**Comment by WitherCubes** [comment]
I installed the nim version from the Arch Repo, I didn't see that it was an old version. Sorry if I wasted anyone's time.

**Comment by codic12** [comment]
Indeed, fixed the version, thanks!

**Comment by codic12** [comment]
i've also fixed compilation with 1.4.8 and brought down minimum version again :+1:
